### ISSUE TITLE
Bundled output on Unix-like systems fail to load a web application in the browser

### ISSUE THREAD

**Issue by miqid** [webpack/webpack#1762]
Howdy there.

I've got a bit of a _seemingly_ esoteric platform-related error with creating a `bundle.js` using Webpack. I'm hoping that somebody recognises this as being a silly blunder I might have done with configuration. On my Windows machine when I run up Webpack against my configuration file, I can load the bundled output in a browser. However, on any Unix-like system, the bundled output when loaded in a browser produces a `require is not defined` error (snippet below makes this obvious). This is appears to be coming from module chunks mainly involving style-loader (`imports-loader` as well) - though this might be a red herring.

I did a diff between the `bundle.js` files produced by each platform and I noticed the ones churned out by the Unix-like systems have a module chunk which looks like:
```js
...
/***/ },
/* 77 */
/***/ function(module, exports) {
// style-loader: Adds some css to the DOM by adding a <style> tag
// load the styles
var content = require("!!./../../node_modules/css-loader/index.js!./animate.css");
if(typeof content === 'string') content = [[module.id, content, '']];
// add the styles to the DOM
var update = require("!./../../node_modules/style-loader/addStyles.js")(content, {});
if(content.locals) module.exports = content.locals;
...
```
Whereas locating a similar-looking chunk in the Windows version looks like:
```js
...
/***/ },
/* 77 */
/***/ function(module, exports, __webpack_require__) {
// style-loader: Adds some css to the DOM by adding a <style> tag
// load the styles
var content = __webpack_require__(78);
if(typeof content === 'string') content = [[module.id, content, '']];
// add the styles to the DOM
var update = __webpack_require__(80)(content, {});
...
```
I've already tried re-installing all my project dependencies via npm too.

And the relevant snippet from my Webpack configuration:
```js
module: {
  noParse: [],
  loaders: [
    { test: /[\\\/]bootstrap(\.min)?\.js$/, loader: 'imports?jQuery=jquery'},
    { test: /\.html$/, loader: 'html' },
    { test: /\.css$/, loader: 'style!css' },
    { test: /\.(scss|sass)$/, loader: 'style!css!sass' },
    { test: /.*/, loader: 'file?name=assets/[name].[ext]&context=./src/app/',
      include: [
        /assets[\\\/]images.+/,
        /assets[\\\/]data.+/
      ]
    },
    { test: /\.(png|woff|woff2|eot|ttf|svg)$/, loader: 'url',
      include: [
        /bower_components.+/,
        /assets[\\\/]fonts.+/
      ]
    }
  ]
}
```

Any advice as to why Webpack might not be transforming those `require` calls?

**Comment by sokra** [comment]
Looks like an incorrect `noParse` configuration.

**Comment by miqid** [comment]
Turned out to be a configuration blunder on my behalf.

I was populating the `noParse` array with some code further on from the main Webpack configuration object, but doing it in a way that apparently didn't play well with the `UglifyJsPlugin`. After a bit of tweaking, I was able to get bundled distributions churned out across all platforms without problem. It still puzzles me why the bundling was working when I built with Windows to begin with.

@sokra, thank you for the helpful hint.
